retrieval/discovery/zookeeper.go

```diff
@@ -0,0 +1,213 @@
+package discovery
```

Comment:
I meant to put this under the `util` directory. This is the only Go implementation of a Zookeeper TreeCache, and at least one person was looking to use it elsewhere.